src/Document/Metadata/MetadataInterface.php

```diff
@@ -68,4 +68,14 @@ public function getSortableAttributes(): array;
      * @return list<string>
      */
     public function getSortableAttributeNames(): array;
+
+    /**
+     * @return array<string, list<string>>
+     */
+    public function getMapProductOptions(): array;
```

Comment:
```suggestion
    public function getMappedProductOptions(): array;
```

Reply:
I reckon we also need a comment for both of these new methods, don't you think? :D I know you're not a fan, but maybe one that isn't obvious 😂

Reply:
I've added the comment (even though indeed I do not feel it's needed xD)